lib/compress/zstd_compress_internal.h

```diff
@@ -745,27 +745,27 @@ ZSTD_count_2segments(const BYTE* ip, const BYTE* match,
  *  Hashes
  ***************************************/
 static const U32 prime3bytes = 506832829U;
-static U32    ZSTD_hash3(U32 u, U32 h) { return ((u << (32-24)) * prime3bytes)  >> (32-h) ; }
+static U32    ZSTD_hash3(U32 u, U32 h) { assert(h <= 32); return ((u << (32-24)) * prime3bytes)  >> (32-h) ; }
```

Comment:
I think there is a good reason to leave them. It validates that `SHORT_CACHE_TAG_BITS + mls` is not too large (in case someone decides to increase it in the future).

The main reason I can see to take them out would be debug performance. Is that why you think we should delete them? Or a different reason?

Reply:
Indeed my concern is debug performance. Maybe that's not an important concern.

Reply:
I'm not sure how to value the safety win against the debug perf hit. @Cyan4973 can you weigh in?

Reply:
I'm not concerned about debug performance. We've never considered it important so far.

Reply:
Oh, I just realized that `h` is known at compile time anyway :)

Reply:
Agreed, such a simple `assert()` is fine with regards to debug performance.
Plus, if `h` is static and the compiler can understand it, it will probably remove the `assert()` anyway.